PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/FunctionDeploymentPanel.java

```diff
@@ -84,15 +248,118 @@ protected JLabel getLblMavenProject() {
 
     @Override
     protected void resetFromConfig(@NotNull FunctionDeployConfiguration configuration) {
-
+        configuration.setFunctionId(configuration.getFunctionId());
+        final Module previousModule = configuration.getModule();
+        if (previousModule != null) {
+            for (int i = 0; i < cbFunctionModule.getItemCount(); i++) {
+                final Module module = cbFunctionModule.getItemAt(i);
+                if (Paths.get(module.getModuleFilePath()).equals(Paths.get(previousModule.getModuleFilePath()))) {
+                    cbFunctionModule.setSelectedIndex(i);
+                    break;
+                }
+            }
+        }
+        presenter.loadFunctionApps(false);
     }
 
     @Override
     protected void apply(@NotNull FunctionDeployConfiguration configuration) {
+        if (selectedFunctionApp == null || selectedFunctionApp.getResource() == null) {
+            return;
+        }
+        final FunctionApp targetFunction = selectedFunctionApp.getResource();
+        configuration.setAppName(targetFunction.name());
+        configuration.setResourceGroup(targetFunction.resourceGroupName());
+        configuration.setSubscription(selectedFunctionApp.getSubscriptionId());
+        configuration.setFunctionId(targetFunction.id());
+
+        final IntelliJFunctionRuntimeConfiguration runtimeConfiguration = new IntelliJFunctionRuntimeConfiguration();
+        final AppServicePlan appServicePlan = getAppServicePlan(targetFunction);
+        runtimeConfiguration.setOs(appServicePlan.operatingSystem() == OperatingSystem.WINDOWS ? "windows" : "linux");
+        configuration.setRuntime(runtimeConfiguration);
+
+        configuration.setPricingTier(appServicePlan.pricingTier().toSkuDescription().size());
+        configuration.setAppSettings(appSettingsTable.getAppSettings());
+        configuration.setModule((Module) cbFunctionModule.getSelectedItem());
```

Comment:
shall we use RunConfigurationModule? see ModuleBasedConfiguration

Reply:
`AzureRunConfigurationBase` is based on `RunConfigurationBase`, and other services like app services depends on project for now, we could investigate whether we could move to `ModuleBasedConfiguration` later